controllers/challenges.js

```diff
@@ -7,12 +11,42 @@ const {StatusCodes} = require('http-status-codes');
  * @url GET /challenges 
  * @query count - number of past challenges to return
  * @response challenges - list of current and past challenges
- * @deprecated not implemented
  */
 const getChallenges = async (req, res) => {
+    let {count} = req.query;
+    const{ user:userID } = req.body;
+    const user = await User.findById(userID).populate({
+        path:'challenges',
+        mode:'Challenge',
+        select:'_id title description startDate endDate difficulty status'
+    }).limit(count);
```

Comment:
please change the way you are limiting the amount of challenges returned. We always want to return the current challenges and count variable should only impact past ones.